sklearn/inspection/_partial_dependence.py

```diff
@@ -88,17 +89,26 @@ def _grid_from_X(X, percentiles, is_categorical, grid_resolution):
 
     values = []
     for feature, is_cat in enumerate(is_categorical):
-        uniques = np.unique(_safe_indexing(X, feature, axis=1))
+        uniques = _unique(_safe_indexing(X, feature, axis=1))
+        uniques = np.array(
+            [value for value in uniques if not is_scalar_nan(value)],
+            dtype=uniques.dtype,
+        )
```

Comment:
IIUC, `_unique` will return at maximum one nan value at the end of the array such that the following would work:
```suggestion
if is_scalar_nan(uniques[-1]):
    uniques = uniques[:-1]
```